File: scripts/_build_intro_deck.js
/**
 * scripts/_build_intro_deck.py 대응 노드 스크립트 — K1a 카카오 연동대행사 신청 소개서.
 *
 * 디자인 v3 토큰 승계(app.css :root): 먹 #1A1714 · 한지 #F5EFE3 · 종이 #FBF8F1 ·
 * 금 #C9A24B · 청록 #119A8E · 주황 #F5821F(강조 1점) · 라인 #E6DECB.
 * "디지털 한지 위의 금속활자" — 어두운 표지·마무리, 밝은 본문(샌드위치).
 *
 * 사실만 담는다. 없는 수치는 [오너 기입]. 발명 0.
 */
const pptx = require("pptxgenjs");
const fs = require("fs");

// ★ 콘텐츠 단일 소스 — PDF 생성기(build_intro_deck.py)와 **같은 JSON**을 읽는다.
//   문구를 두 곳에 두면 한쪽만 고쳐진다(이번 세션 최다 결함).
const C = JSON.parse(fs.readFileSync("docs/apply/intro_content.json", "utf-8"));

const INK = "1A1714", INK2 = "2A241E", HANJI = "F5EFE3", PAPER = "FBF8F1";
const GOLD = "C9A24B", GOLDSOFT = "E0C588", TEAL = "119A8E", ORANGE = "F5821F";
const LINE = "E6DECB", MUTED = "8A8275", INKSOFT = "3A352E";
const SERIF = "Cambria", SANS = "Calibri";

const p = new pptx();
p.layout = "LAYOUT_WIDE";                 // 13.3 x 7.5
const W = 13.3, H = 7.5, M = 0.75;

const notes = (s, t) => s.addNotes(t);

/** 어두운 슬라이드 공통 바탕(표지·마무리). */
function darkSlide() {
  const s = p.addSlide();
  s.background = { color: INK };
  return s;
}
/** 밝은 본문 슬라이드 + 제목. */
function lightSlide(kicker, title) {
  const s = p.addSlide();
  s.background = { color: PAPER };
  s.addText(kicker, {
    x: M, y: 0.45, w: 8, h: 0.3, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 11, bold: true, charSpacing: 2, color: GOLD,
  });
  s.addText(title, {
    x: M, y: 0.8, w: W - M * 2, h: 0.8, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 34, bold: true, color: INK,
  });
  return s;
}
/** 카드 — 배경 톤 + 그림자. 테두리 줄무늬 금지. */
function card(s, x, y, w, h, fill) {
  s.addShape(p.ShapeType.roundRect, {
    x, y, w, h, rectRadius: 0.08,
    fill: { color: fill || "FFFFFF" }, line: { color: LINE, width: 0.75 },
    shadow: { type: "outer", angle: 90, blur: 12, offset: 2, opacity: 0.08, color: INK },
  });
}
/** 번호 원형 배지 — 다리를 건너는 순서(시그니처). */
function stepDot(s, x, y, n, color) {
  s.addShape(p.ShapeType.ellipse, { x, y, w: 0.42, h: 0.42, fill: { color: color || INK } });
  s.addText(String(n), {
    x, y, w: 0.42, h: 0.42, isTextBox: true, margin: 0, align: "center", valign: "middle",
    fontFace: SERIF, fontSize: 15, bold: true, color: "FFFFFF",
  });
}

/* ── 1. 표지 ───────────────────────────────────────────────────────────── */
{
  const s = darkSlide();
  s.addText(C.cover.kicker, {
    x: M, y: 1.5, w: 10, h: 0.35, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 12, bold: true, charSpacing: 3, color: GOLD,
  });
  s.addText(C.cover.brand, {
    x: M, y: 1.95, w: 11, h: 1.3, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 60, bold: true, color: HANJI,
  });
  s.addText(C.cover.tagline, {
    x: M, y: 3.25, w: 11, h: 0.5, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 20, color: GOLDSOFT,
  });
  s.addShape(p.ShapeType.rect, { x: M, y: 4.15, w: 2.2, h: 0.02, fill: { color: GOLD } });

  const rows = C.cover.rows;
  rows.forEach(([k, v], i) => {
    const y = 4.6 + i * 0.42;
    s.addText(k, {
      x: M, y, w: 1.9, h: 0.35, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 12, color: MUTED,
    });
    s.addText(v, {
      x: M + 2.0, y, w: 7, h: 0.35, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 14, color: HANJI,
    });
  });
  notes(s, "신청 명의는 고가네(개인사업자). alaz ltd·우주대행 명의는 사용하지 않는다.");
}

/* ── 2. 서비스 개요 ────────────────────────────────────────────────────── */
{
  const s = lightSlide(C.overview.kicker, C.overview.title);
  const steps = C.overview.steps;
  const cw = 1.85, gap = 0.16;
  steps.forEach(([t, d], i) => {
    const x = M + i * (cw + gap);
    card(s, x, 2.05, cw, 2.15, "FFFFFF");
    stepDot(s, x + 0.2, 2.25, i + 1, i === steps.length - 1 ? ORANGE : INK);
    s.addText(t, {
      x: x + 0.2, y: 2.82, w: cw - 0.4, h: 0.35, isTextBox: true, margin: 0,
      fontFace: SERIF, fontSize: 15, bold: true, color: INK,
    });
    s.addText(d, {
      x: x + 0.2, y: 3.2, w: cw - 0.4, h: 0.85, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 10, color: INKSOFT, lineSpacing: 14,
    });
  });
  s.addText(C.overview.note, {
    x: M, y: 4.45, w: W - M * 2, h: 0.4, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 14, color: INKSOFT,
  });
  s.addImage({
    path: C.overview.shot,
    x: M, y: 5.0, w: 11.8, h: 1.95, sizing: { type: "crop", w: 11.8, h: 1.95, x: 0, y: 0 },
  });
  s.addText(C.overview.shot_caption, {
    x: M, y: 6.98, w: 5, h: 0.28, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 10, color: MUTED,
  });
  notes(s, "화면은 실제 운영 화면 캡처. 목업 아님.");
}

/* ── 3. 연동 현황 ──────────────────────────────────────────────────────── */
{
  const s = lightSlide(C.markets.kicker, C.markets.title);
  const items = C.markets.rows;
  const DOTC = { teal: TEAL, gold: GOLD, orange: ORANGE };   // JSON은 색 이름, pptx는 hex
  items.forEach(([name, api, state, dotKey], i) => {
    const dot = DOTC[dotKey] || TEAL;
    const y = 2.05 + i * 1.02;
    card(s, M, y, 11.8, 0.88, "FFFFFF");
    s.addShape(p.ShapeType.ellipse, { x: M + 0.35, y: y + 0.33, w: 0.22, h: 0.22,
                                      fill: { color: dot } });
    s.addText(name, {
      x: M + 0.75, y: y + 0.16, w: 2.4, h: 0.3, isTextBox: true, margin: 0,
      fontFace: SERIF, fontSize: 17, bold: true, color: INK,
    });
    s.addText(api, {
      x: M + 0.75, y: y + 0.48, w: 2.4, h: 0.28, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 11, color: MUTED,
    });
    s.addText(state, {
      x: M + 3.4, y: y + 0.28, w: 8.0, h: 0.35, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 14, color: INKSOFT,
    });
  });
  s.addText(C.markets.note, {
    x: M, y: 6.35, w: W - M * 2, h: 0.35, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 12, italic: true, color: MUTED,
  });
}

/* ── 4. 보안·개인정보 ─────────────────────────────────────────────────── */
{
  const s = lightSlide(C.security.kicker, C.security.title);
  const items = C.security.items;
  items.forEach(([t, d], i) => {
    const col = i % 2, row = Math.floor(i / 2);
    const x = M + col * 6.05, y = 2.05 + row * 1.42;
    const w = i === 4 ? 11.8 : 5.75;
    card(s, x, y, w, 1.22, "FFFFFF");
    s.addShape(p.ShapeType.ellipse, { x: x + 0.28, y: y + 0.28, w: 0.34, h: 0.34,
                                      fill: { color: HANJI } });
    s.addShape(p.ShapeType.ellipse, { x: x + 0.38, y: y + 0.38, w: 0.14, h: 0.14,
                                      fill: { color: TEAL } });
    s.addText(t, {
      x: x + 0.78, y: y + 0.2, w: w - 1.05, h: 0.32, isTextBox: true, margin: 0,
      fontFace: SERIF, fontSize: 15, bold: true, color: INK,
    });
    s.addText(d, {
      x: x + 0.78, y: y + 0.54, w: w - 1.05, h: 0.6, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 11, color: INKSOFT, lineSpacing: 15,
    });
  });
  notes(s, "여기 적은 다섯 가지는 전부 코드에 구현돼 있는 것. 없는 기능은 쓰지 않는다.");
}

/* ── 5. 카카오쇼핑 연동 계획 ──────────────────────────────────────────── */
{
  const s = lightSlide(C.plan.kicker, C.plan.title);
  card(s, M, 2.0, 5.75, 2.5, "FFFFFF");
  s.addText("매핑 모델", {
    x: M + 0.35, y: 2.22, w: 5, h: 0.32, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 17, bold: true, color: INK,
  });
  C.plan.model
    .forEach(([t, d], i) => {
      const y = 2.68 + i * 0.6;
      s.addText(t, {
        x: M + 0.35, y, w: 5.05, h: 0.26, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 12, bold: true, color: TEAL,
      });
      s.addText(d, {
        x: M + 0.35, y: y + 0.26, w: 5.05, h: 0.3, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 10, color: INKSOFT,
      });
    });

  card(s, M + 6.05, 2.0, 5.75, 2.5, "FFFFFF");
  s.addText("연동 범위", {
    x: M + 6.4, y: 2.22, w: 5, h: 0.32, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 17, bold: true, color: INK,
  });
  C.plan.scope
    .forEach(([t, d], i) => {
      const y = 2.68 + i * 0.6;
      s.addText(t, {
        x: M + 6.4, y, w: 5.05, h: 0.26, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 12, bold: true, color: TEAL,
      });
      s.addText(d, {
        x: M + 6.4, y: y + 0.26, w: 5.05, h: 0.3, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 10, color: INKSOFT,
      });
    });

  card(s, M, 4.7, 11.8, 1.35, HANJI);
  C.plan.facts
    .forEach(([k, v], i) => {
      const x = M + 0.4 + i * 3.85;
      s.addText(k, {
        x, y: 4.92, w: 3.6, h: 0.28, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 11, bold: true, charSpacing: 1, color: GOLD,
      });
      s.addText(v, {
        x, y: 5.22, w: 3.6, h: 0.6, isTextBox: true, margin: 0,
        fontFace: SANS, fontSize: 13, color: INK,
      });
    });
  s.addText(C.plan.note, {
    x: M, y: 6.25, w: W - M * 2, h: 0.35, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 12, italic: true, color: MUTED,
  });
}

/* ── 6. 판매자 현황 ───────────────────────────────────────────────────── */
{
  const s = lightSlide(C.sellers.kicker, C.sellers.title);
  const stats = C.sellers.stats;
  stats.forEach(([n, k, d], i) => {
    const x = M + i * 4.0;
    card(s, x, 2.05, 3.75, 1.9, "FFFFFF");
    s.addText(n, {
      x: x + 0.35, y: 2.2, w: 3.0, h: 0.95, isTextBox: true, margin: 0,
      fontFace: SERIF, fontSize: 54, bold: true, color: i === 0 ? ORANGE : INK,
    });
    s.addText(k, {
      x: x + 0.35, y: 3.15, w: 3.0, h: 0.3, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 13, bold: true, color: INK,
    });
    s.addText(d, {
      x: x + 0.35, y: 3.45, w: 3.0, h: 0.3, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 10, color: MUTED,
    });
  });
  card(s, M, 4.2, 11.8, 0.85, HANJI);
  s.addText(C.sellers.headline, {
    x: M + 0.4, y: 4.42, w: 11.0, h: 0.42, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 17, bold: true, color: INK,
  });
  s.addText(C.sellers.goal, {
    x: M, y: 5.25, w: 6, h: 0.35, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 13, color: INKSOFT,
  });
  s.addImage({
    path: C.sellers.shot,
    x: M, y: 5.7, w: 11.8, h: 1.3, sizing: { type: "crop", w: 11.8, h: 1.3, x: 0, y: 0 },
  });
  s.addText(C.sellers.shot_caption, {
    x: M, y: 7.02, w: 5, h: 0.28, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 10, color: MUTED,
  });
  notes(s, "수치는 사실만. 목표는 오너가 채운다 — 부풀리지 않는다.");
}

/* ── 7. 연락처 ────────────────────────────────────────────────────────── */
{
  const s = darkSlide();
  s.addText(C.contact.kicker, {
    x: M, y: 1.6, w: 8, h: 0.35, isTextBox: true, margin: 0,
    fontFace: SANS, fontSize: 12, bold: true, charSpacing: 3, color: GOLD,
  });
  s.addText(C.contact.title, {
    x: M, y: 2.05, w: 11, h: 1.0, isTextBox: true, margin: 0,
    fontFace: SERIF, fontSize: 44, bold: true, color: HANJI,
  });
  s.addShape(p.ShapeType.rect, { x: M, y: 3.25, w: 2.2, h: 0.02, fill: { color: GOLD } });
  const rows = C.contact.rows;
  rows.forEach(([k, v], i) => {
    const y = 3.7 + i * 0.5;
    s.addText(k, {
      x: M, y, w: 2.2, h: 0.38, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 12, color: MUTED,
    });
    s.addText(v, {
      x: M + 2.3, y, w: 8, h: 0.38, isTextBox: true, margin: 0,
      fontFace: SANS, fontSize: 16, color: HANJI,
    });
  });
  s.addText("고가브릿지", {
    x: W - 4.2, y: 6.4, w: 3.5, h: 0.5, isTextBox: true, margin: 0, align: "right",
    fontFace: SERIF, fontSize: 22, bold: true, color: GOLDSOFT,
  });
}

p.writeFile({ fileName: "docs/apply/gogabridj_intro_kakao_v1.pptx" })
  .then((f) => console.log("saved", f));

File: scripts/build_intro_deck.py
"""scripts/build_intro_deck.py — K1a 소개서 **PDF** 생성(Chromium 인쇄).

`docs/apply/intro_content.json` **하나가 콘텐츠 단일 소스**다. pptx(편집용)와 PDF(제출용)는
같은 JSON을 읽는 **두 개의 출력기**일 뿐 — 문구가 두 벌로 갈리지 않는다(계약이 검사).

**왜 Chromium인가:** 이 샌드박스의 LibreOffice가 pptx를 하나도 못 연다(빈 python-pptx
파일로도 재현 — `Error: source file could not be loaded`). 그래서 soffice → PDF 경로가 없다.
대신 우리 실제 디자인 토큰(app.css)을 그대로 쓰는 HTML을 Chromium이 인쇄한다 —
브랜드 재현은 오히려 이쪽이 정확하다.

    python scripts/build_intro_deck.py
"""
from __future__ import annotations

import base64
import json
import mimetypes
import os
import sys

ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
CONTENT = os.path.join(ROOT, "docs/apply/intro_content.json")
OUT_PDF = os.path.join(ROOT, "docs/apply/gogabridj_intro_kakao_v2.pdf")
CHROME = "/opt/pw-browsers/chromium-1194/chrome-linux/chrome"

# 디자인 v3 토큰(app.css :root) 승계 — 값을 여기서 새로 만들지 않는다.
T = {"ink": "#1A1714", "ink2": "#2A241E", "hanji": "#F5EFE3", "paper": "#FBF8F1",
     "gold": "#C9A24B", "goldsoft": "#E0C588", "teal": "#119A8E", "orange": "#F5821F",
     "line": "#E6DECB", "muted": "#8A8275", "inksoft": "#3A352E"}


def _img(rel_path: str) -> str:
    """이미지를 data URI로 인라인 — 인쇄 시 외부 로드에 의존하지 않게."""
    path = os.path.join(ROOT, rel_path)
    if not os.path.exists(path):
        return ""
    mime = mimetypes.guess_type(path)[0] or "image/png"
    with open(path, "rb") as fp:
        return f"data:{mime};base64," + base64.b64encode(fp.read()).decode()


def _esc(s) -> str:
    return (str(s).replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;"))


def build_html(c: dict) -> str:
    cov, ov, mk, sec, pl, sl, ct = (c["cover"], c["overview"], c["markets"], c["security"],
                                    c["plan"], c["sellers"], c["contact"])

    def kicker(k):
        return f'<div class="kicker">{_esc(k)}</div>'

    def head(blk):
        return kicker(blk["kicker"]) + f'<h2>{_esc(blk["title"])}</h2>'

    steps = "".join(
        f'<div class="step"><div class="dot{" accent" if i == 5 else ""}">{i + 1}</div>'
        f'<div class="st">{_esc(t)}</div><div class="sd">{_esc(d)}</div></div>'
        for i, (t, d) in enumerate(ov["steps"]))

    mkrows = "".join(
        f'<div class="row"><span class="bullet {cls}"></span>'
        f'<div class="rn"><b>{_esc(n)}</b><span>{_esc(api)}</span></div>'
        f'<div class="rs">{_esc(st)}</div></div>'
        for n, api, st, cls in mk["rows"])

    secitems = "".join(
        f'<div class="sec-card{" wide" if i == 4 else ""}"><span class="ring"></span>'
        f'<b>{_esc(t)}</b><p>{_esc(d)}</p></div>'
        for i, (t, d) in enumerate(sec["items"]))

    def pairs(items):
        return "".join(f'<div class="pair"><b>{_esc(t)}</b><span>{_esc(d)}</span></div>'
                       for t, d in items)

    facts = "".join(f'<div class="fact"><span>{_esc(k)}</span><b>{_esc(v)}</b></div>'
                    for k, v in pl["facts"])
    stats = "".join(
        f'<div class="stat"><div class="num{" accent" if i == 0 else ""}">{_esc(n)}</div>'
        f'<div class="sk">{_esc(k)}</div><div class="sm">{_esc(d)}</div></div>'
        for i, (n, k, d) in enumerate(sl["stats"]))

    def rowtable(rows, cls=""):
        return "".join(f'<div class="kv {cls}"><span>{_esc(k)}</span><b>{_esc(v)}</b></div>'
                       for k, v in rows)

    return f"""<!doctype html><html lang="ko"><head><meta charset="utf-8">
<style>
  @page {{ size: 330mm 176mm; margin: 0; }}
  * {{ box-sizing: border-box; }}
  body {{ margin: 0; font-family: "Noto Sans KR", "Malgun Gothic", sans-serif;
          color: {T['ink']}; -webkit-print-color-adjust: exact; print-color-adjust: exact; }}
  .slide {{ width: 330mm; height: 176mm; padding: 18mm 20mm; page-break-after: always;
            position: relative; overflow: hidden; background: {T['paper']}; }}
  .slide:last-child {{ page-break-after: auto; }}
  .dark {{ background: {T['ink']}; color: {T['hanji']}; }}
  .kicker {{ font-size: 10pt; font-weight: 700; letter-spacing: .22em; color: {T['gold']};
             margin-bottom: 4mm; }}
  h1 {{ font-family: Georgia, "Times New Roman", serif; font-size: 46pt; margin: 0 0 3mm;
        letter-spacing: -.02em; }}
  h2 {{ font-family: Georgia, "Times New Roman", serif; font-size: 27pt; margin: 0 0 8mm;
        letter-spacing: -.015em; }}
  .tagline {{ font-size: 15pt; color: {T['goldsoft']}; margin-bottom: 8mm; }}
  .rule {{ width: 42mm; height: .6mm; background: {T['gold']}; margin: 0 0 9mm; }}
  .kv {{ display: flex; gap: 8mm; padding: 2.2mm 0; font-size: 11pt; }}
  .kv span {{ width: 34mm; color: {T['muted']}; }}
  .kv b {{ font-weight: 500; }}
  .steps {{ display: grid; grid-template-columns: repeat(6, 1fr); gap: 4mm; margin-bottom: 7mm; }}
  .step {{ background: #fff; border: .3mm solid {T['line']}; border-radius: 2.5mm; padding: 5mm; }}
  .dot {{ width: 9mm; height: 9mm; border-radius: 50%; background: {T['ink']}; color: #fff;
          font-family: Georgia, serif; font-weight: 700; font-size: 12pt; text-align: center;
          line-height: 9mm; margin-bottom: 4mm; }}
  .dot.accent {{ background: {T['orange']}; }}
  .st {{ font-family: Georgia, serif; font-weight: 700; font-size: 12.5pt; margin-bottom: 2mm; }}
  .sd {{ font-size: 9pt; color: {T['inksoft']}; line-height: 1.55; }}
  .note {{ font-size: 11pt; color: {T['inksoft']}; margin-bottom: 6mm; }}
  .muted-note {{ font-size: 10pt; color: {T['muted']}; font-style: italic; }}
  .shot {{ width: 100%; border: .3mm solid {T['line']}; border-radius: 2mm; object-fit: cover;
           object-position: top center; }}
  .cap {{ font-size: 8.5pt; color: {T['muted']}; margin-top: 2mm; }}
  .row {{ display: flex; align-items: center; gap: 6mm; background: #fff; border-radius: 2.5mm;
          border: .3mm solid {T['line']}; padding: 5mm 7mm; margin-bottom: 4mm; }}
  .bullet {{ width: 4mm; height: 4mm; border-radius: 50%; flex: none; }}
  .bullet.teal {{ background: {T['teal']}; }}  .bullet.gold {{ background: {T['gold']}; }}
  .rn {{ width: 52mm; }}
  .rn b {{ font-family: Georgia, serif; font-size: 13pt; display: block; }}
  .rn span {{ font-size: 9pt; color: {T['muted']}; }}
  .rs {{ font-size: 12pt; color: {T['inksoft']}; }}
  .sec-grid {{ display: grid; grid-template-columns: 1fr 1fr; gap: 5mm; }}
  .sec-card {{ background: #fff; border: .3mm solid {T['line']}; border-radius: 2.5mm;
               padding: 5mm 6mm; position: relative; }}
  .sec-card.wide {{ grid-column: 1 / -1; }}
  .ring {{ position: absolute; left: 6mm; top: 6mm; width: 5mm; height: 5mm; border-radius: 50%;
           background: {T['hanji']}; box-shadow: inset 0 0 0 1.4mm {T['teal']}; }}
  .sec-card b {{ display: block; margin-left: 9mm; font-family: Georgia, serif; font-size: 12.5pt; }}
  .sec-card p {{ margin: 2mm 0 0 9mm; font-size: 9.5pt; color: {T['inksoft']}; line-height: 1.6; }}
  .two {{ display: grid; grid-template-columns: 1fr 1fr; gap: 5mm; margin-bottom: 5mm; }}
  .panel {{ background: #fff; border: .3mm solid {T['line']}; border-radius: 2.5mm; padding: 6mm; }}
  .panel h3 {{ font-family: Georgia, serif; font-size: 14pt; margin: 0 0 4mm; }}
  .pair {{ margin-bottom: 3.5mm; }}
  .pair b {{ display: block; font-size: 10.5pt; color: {T['teal']}; }}
  .pair span {{ font-size: 9pt; color: {T['inksoft']}; }}
  .facts {{ display: grid; grid-template-columns: repeat(3, 1fr); gap: 6mm;
            background: {T['hanji']}; border-radius: 2.5mm; padding: 6mm 7mm; margin-bottom: 5mm; }}
  .fact span {{ display: block; font-size: 9pt; font-weight: 700; letter-spacing: .1em;
                color: {T['gold']}; margin-bottom: 1.5mm; }}
  .fact b {{ font-size: 11.5pt; font-weight: 500; }}
  .stats {{ display: grid; grid-template-columns: repeat(3, 1fr); gap: 6mm; margin-bottom: 5mm; }}
  .stat {{ background: #fff; border: .3mm solid {T['line']}; border-radius: 2.5mm; padding: 6mm; }}
  .num {{ font-family: Georgia, serif; font-size: 40pt; font-weight: 700; line-height: 1; }}
  .num.accent {{ color: {T['orange']}; }}
  .sk {{ font-size: 11pt; font-weight: 700; margin-top: 3mm; }}
  .sm {{ font-size: 9pt; color: {T['muted']}; }}
  .headline {{ background: {T['hanji']}; border-radius: 2.5mm; padding: 5mm 7mm;
               font-family: Georgia, serif; font-size: 14pt; font-weight: 700; margin-bottom: 4mm; }}
  .dark .kv span {{ color: {T['muted']}; }}
  .dark .kv b {{ color: {T['hanji']}; }}
  .wordmark {{ position: absolute; right: 20mm; bottom: 16mm; font-family: Georgia, serif;
               font-size: 18pt; font-weight: 700; color: {T['goldsoft']}; }}
</style></head><body>

<section class="slide dark">
  {kicker(cov['kicker'])}
  <h1>{_esc(cov['brand'])}</h1>
  <div class="tagline">{_esc(cov['tagline'])}</div>
  <div class="rule"></div>
  {rowtable(cov['rows'])}
</section>

<section class="slide">
  {head(ov)}
  <div class="steps">{steps}</div>
  <div class="note">{_esc(ov['note'])}</div>
  <img class="shot" style="height:56mm" src="{_img(ov['shot'])}" alt="">
  <div class="cap">{_esc(ov['shot_caption'])}</div>
</section>

<section class="slide">
  {head(mk)}
  {mkrows}
  <div class="muted-note">{_esc(mk['note'])}</div>
</section>

<section class="slide">
  {head(sec)}
  <div class="sec-grid">{secitems}</div>
</section>

<section class="slide">
  {head(pl)}
  <div class="two">
    <div class="panel"><h3>매핑 모델</h3>{pairs(pl['model'])}</div>
    <div class="panel"><h3>연동 범위</h3>{pairs(pl['scope'])}</div>
  </div>
  <div class="facts">{facts}</div>
  <div class="muted-note">{_esc(pl['note'])}</div>
</section>

<section class="slide">
  {head(sl)}
  <div class="stats">{stats}</div>
  <div class="headline">{_esc(sl['headline'])}</div>
  <div class="note">{_esc(sl['goal'])}</div>
  <img class="shot" style="height:38mm" src="{_img(sl['shot'])}" alt="">
  <div class="cap">{_esc(sl['shot_caption'])}</div>
</section>

<section class="slide dark">
  {kicker(ct['kicker'])}
  <h1 style="font-size:36pt">{_esc(ct['title'])}</h1>
  <div class="rule"></div>
  {rowtable(ct['rows'])}
  <div class="wordmark">고가브릿지</div>
</section>

</body></html>"""


def main() -> int:
    with open(CONTENT, encoding="utf-8") as fp:
        content = json.load(fp)
    html = build_html(content)
    html_path = "/tmp/_intro_deck.html"
    with open(html_path, "w", encoding="utf-8") as fp:
        fp.write(html)

    from playwright.sync_api import sync_playwright
    with sync_playwright() as pw:
        br = pw.chromium.launch(executable_path=CHROME)
        pg = br.new_page()
        pg.goto(f"file://{html_path}")
        pg.wait_for_timeout(900)
        pg.pdf(path=OUT_PDF, width="330mm", height="176mm", print_background=True,
               margin={"top": "0", "bottom": "0", "left": "0", "right": "0"})
        pg.close()
        br.close()

    size = os.path.getsize(OUT_PDF)
    print(f"저장 {OUT_PDF} · {size:,} bytes ({size / 1_048_576:.2f} MB)")
    if size > 10 * 1_048_576:
        print("⚠ 10MB 초과 — 이미지 압축 필요")
        return 1
    return 0


if __name__ == "__main__":
    sys.exit(main())
